front/src/modules/ui/themes/hooks/useSystemColorScheme.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { ColorScheme } from '../states/colorSchemeState';
+
+export type SystemColorScheme = ColorScheme.Light | ColorScheme.Dark;
+
+export function useSystemColorScheme(): SystemColorScheme {
```

Comment:
I don't think this can live into ui module. We will need to get information for the user. Therefore, this would create a dependency on User in a UI module which we want to avoid!
UI components should not know about business entities. Maybe move this hook to user module

Reply:
Actually the hook `useSystemColorScheme` is totally based on system, and theme module is based on system too, so for that one I guess it's the correct place.
I can move the hook `useColorScheme` into user module as this one call `updateUser` from back-end, but I don't think putting this two hooks in different folders is a good option.
`colorScheme` is only used for theming purpose, we need to figure out if we create module based on their need or based on their use. For me the second option is more clear.